google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -297,50 +347,156 @@ def from_gcs_prefix_with_revisions(self, gcs_prefix: str):
         return root_revision_nodes[-1]
 
     def last_revision(self):
+        r"""Goes to the previous revision.
+
+            This should be used if you want to go to the previous revision.
+
+            For example: Given revision tree with current revision being 3
+            ```
+                └──1
+                    └──2
+                      └──>3
+                    └──5
+                ├──4
+            ```
+            If you use `next_revision()` the next revision will be 2
+
+        Args:
+            None
+        Returns:
+            DocumentWithRevisions:
+                The next DocumentWithRevision object.
+        """
         if self.parent:
             current_index = self.parent.children_ids.index(self.revision_id)
             if current_index != 0:
                 return self.parent.children[current_index - 1]
-            elif current_index != None:
-                print("hi")
+            elif current_index is not None:
                 return self.parent
+        elif self.revision_id in self.parent_ids:
+            non_parent_index = self.parent_ids.index(self.revision_id)
+            if non_parent_index > 0:
+                next_parent = self.parent_ids[non_parent_index - 1]
+                index = self.all_node_ids.index(next_parent)
+                return self.root_revision_nodes[index]
+            else:
+                return self
         return self
```

Comment:
I think this if/else tree could be simplified to make it easier to read. I find that "guard" statements work well to visualize the flow. Swap the conditions for the `if` and add returns early instead of using many `else`s.